File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        for(int i = 0; i < (matrix.size()+1)/2; i++){//layer
            for(int j = 0; j < (matrix.size()-1); j++){//jth element in layer
                if(j<i || j>(matrix.size()-i-2)){
                    continue;
                }
                cout << "i" << i << "j" << j << endl;
                swap(matrix[i][j], matrix[matrix.size()-j-1][i]);
                swap(matrix[matrix.size()-j-1][i], matrix[matrix.size()-i-1][matrix.size()-j-1]);
                swap(matrix[matrix.size()-i-1][matrix.size()-j-1], matrix[j][matrix.size()-i-1]);
            }
        }
    }
};

File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if i > j:
                    continue
                matrix[i][j], matrix[j][i] = matrix[j][i], matrix[i][j]
        print(matrix)
        for i in range(len(matrix)):
            for j in range(len(matrix[0])):
                if j >= len(matrix[0])/2:
                    continue
                matrix[i][j], matrix[i][len(matrix[0])-j-1] = matrix[i][len(matrix[0])-j-1], matrix[i][j]
        